engine/render/src/render/render.cpp

```diff
@@ -228,6 +225,10 @@ namespace dmRender
             render_context->m_RenderList.SetSize(list_size);
         }
 
+        if (end == begin) {
+            return;
+        }
```

Comment:
We need an opportunity to return the unused entries to the list before we early out.